deploy/deploy.go

```diff
@@ -645,3 +684,39 @@ func neoFSRuntimeTransactionModifier(getBlockchainHeight func() uint32) actor.Tr
 		return nil
 	}
 }
+
+// first is for notary, second is for alphabet.
+func checkCommitteeRoles(b Blockchain, m *blockchainMonitor, committee keys.PublicKeys) (bool, bool, error) {
+	roleContract := rolemgmt.NewReader(invoker.New(b, nil))
+
+	notaryRole, err := checkRole(noderoles.P2PNotary, roleContract, m, committee)
+	if err != nil {
+		return false, false, fmt.Errorf("%s role check: %w", noderoles.P2PNotary, err)
+	}
+
+	alphaRole, err := checkRole(noderoles.NeoFSAlphabet, roleContract, m, committee)
+	if err != nil {
+		return false, false, fmt.Errorf("%s role check: %w", noderoles.NeoFSAlphabet, err)
+	}
+
+	return notaryRole, alphaRole, nil
+}
+
+func checkRole(role noderoles.Role, roleContract *rolemgmt.ContractReader, m *blockchainMonitor, committee keys.PublicKeys) (bool, error) {
+	currentRoles, err := roleContract.GetDesignatedByRole(role, m.currentHeight())
+	if err != nil {
+		return false, fmt.Errorf("reading %s role: %w", role, err)
```

Comment:
ahhh it always kills me when error context contains function input, caller will most likely get tautology

Reply:
overall agree